source/fundamentals/gridfs.txt

```diff
@@ -0,0 +1,707 @@
+
+   * - ``cancellationToken``
+     - | *Optional.* A token that you can use to cancel the operation.
+       | 
+       | **Data type**: `CancellationToken <https://learn.microsoft.com/dotnet/api/system.threading.cancellationtoken>`__
+
+The following code example shows how to retrieve and print file metadata
+from files in a GridFS bucket. The ``Find()`` method returns an
+``IAsyncCursor<GridFSFileInfo>`` instance from
+which you can access the results.It uses a ``foreach`` loop to iterate through
+the returned cursor and display the contents of the files uploaded in the
+:ref:`gridfs-upload-files` examples. 
+
+Select the :guilabel:`Asynchronous` or :guilabel:`Synchronous` tab to see the corresponding
+code.
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: find-async
+
+      .. io-code-block::
+         :copyable:
+
+         .. input:: /includes/code-examples/GridFS.cs
+            :start-after: // start-find-async
+            :end-before: // end-find-async
+            :language: csharp
+            :dedent:
+
+         .. output::
+            :visible: false
+
+            { "_id" : { "$oid" : "..." }, "length" : 13, "chunkSize" : 261120, "uploadDate" :
+            { "$date" : ... }, "filename" : "new_file" }
+            { "_id" : { "$oid" : "..." }, "length" : 50, "chunkSize" : 1048576, "uploadDate" :
+            { "$date" : ... }, "filename" : "my_file" }
+
+   .. tab:: Synchronous
+      :tabid: find-sync
+
+      .. io-code-block::
+         :copyable:
+
+         .. input:: /includes/code-examples/GridFS.cs
+            :start-after: // start-find
+            :end-before: // end-find
+            :language: csharp
+            :dedent:
+
+         .. output::
+            :visible: false
+
+            { "_id" : { "$oid" : "..." }, "length" : 13, "chunkSize" : 261120, "uploadDate" :
+            { "$date" : ... }, "filename" : "new_file" }
+            { "_id" : { "$oid" : "..." }, "length" : 50, "chunkSize" : 1048576, "uploadDate" :
+            { "$date" : ... }, "filename" : "my_file" }
+
+To customize the find operation, pass an instance of the
+``GridFSFindOptions`` class to the ``FindAsync()`` or ``Find()`` method. The
+``GridFSFindOptions`` class contains the following properties:
+
+.. list-table::
+
+   * - Property
+     - Description
+   
+   * - ``Sort``
+     - | The sort order of the results. If you don't specify a sort order, the
+         method returns the results in the order in which they were inserted.
+       | 
+       | **Data type**: ``SortDefinition<GridFSFileInfo>``. For more information, see
+         the API documentation for the
+         `Sort <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.GridFS.GridFSFindOptions.Sort.html>`__ property.
+
+.. _gridfs-delete-files:
+
+Delete Files
+------------
+
+To delete files from a GridFS bucket, call the ``DeleteAsync()`` or ``Delete()`` method
+on your ``GridFSBucket`` instance. This method removes a file's collection document and
```

Comment:
```suggestion
on your ``GridFSBucket`` instance. This method removes a file's metadata document and
```

Reply:
good call on metadata, but the diagram earlier in the page and [method code](https://github.com/mongodb/mongo-csharp-driver/blob/6a33c2748cc702d8f70242d0a3b8c270e2ae05fe/src/MongoDB.Driver/GridFS/GridFSBucket.cs) lead me to believe it's deleting the whole `files` collection. do you know whether that collection always contains just one document?